packages/ember-data/lib/serializers/json-api-serializer.js

```diff
@@ -25,7 +25,7 @@ export default JSONSerializer.extend({
 
     if (Ember.typeOf(documentHash.data) === 'object') {
       documentHash.data = this._normalizeResourceHelper(documentHash.data);
-    } else {
+    } else if (Ember.typeOf(documentHash.data) === 'array') {
```

Comment:
@wecc would you prefer a more readable way without the `else if`, like we do in [`json-api-serializer/extractRelationship`](http://git.io/vtCxf)?